smithy-linters/src/main/java/software/amazon/smithy/linters/MissingSensitiveTraitValidator.java

```diff
@@ -0,0 +1,196 @@
+                    .stream()
+                    .map(phrase -> phrase.toLowerCase(Locale.US))
+                    .collect(Collectors.toSet()));
+            sensitivePhrases = Collections.unmodifiableSet(phrasesInit);
+            sensitiveWords = Collections.unmodifiableSet(DEFAULT_SENSITIVE_WORDS);
+        } else {
+            if (config.getPhrases().isEmpty()) {
+                //This configuration combination makes the validator a no-op.
+                throw new IllegalArgumentException("Cannot set 'excludeDefaults' to true and leave "
+                                                 + "'phrases' empty or unspecified.");
+            }
+            sensitivePhrases = Collections.unmodifiableSet(new HashSet<>(config.getPhrases()));
+            sensitiveWords = Collections.EMPTY_SET;
+        }
+    }
+
+    /**
+     * Finds shapes without the sensitive trait that possibly contain sensitive data,
+     * based on the shape/member name and the list of key words and phrases.
+     * @param model Model to validate.
+     * @return list of violation events
+     */
+    @Override
+    public List<ValidationEvent> validate(Model model) {
+        List<ValidationEvent> validationEvents = new ArrayList<>();
+        validationEvents.addAll(scanShapeNames(model));
+        validationEvents.addAll(scanMemberNames(model));
+        return validationEvents;
+    }
+
+    private List<ValidationEvent> scanShapeNames(Model model) {
+        List<ValidationEvent> validationEvents = new ArrayList<>();
+        model.shapes()
+                .filter(shape -> !shape.isMemberShape()
+                        && !shape.isOperationShape()
+                        && !shape.isServiceShape()
+                        && !shape.isResourceShape())
+                .filter(shape -> !shape.hasTrait(SENSITIVE))
+                .filter(shape -> containsSensitiveWord(shape.toShapeId().getName())
+                        || containsSensitivePhrase(shape.toShapeId().getName()))
+                .forEach(shape -> validationEvents.add(emit(shape)));
+
+        return validationEvents;
+    }
+
+    private List<ValidationEvent> scanMemberNames(Model model) {
+        return model.shapes()
+                // filter out members with an already sensitive enclosing shape
+                .filter(shape -> !shape.hasTrait(SENSITIVE))
+                .flatMap(shape -> shape.members().stream())
+                // filter out members that target a sensitive shape
+                .filter(memberShape ->
+                        model.getShape(memberShape.getTarget())
+                        .map(shape -> !shape.hasTrait(SENSITIVE))
+                        .orElse(false))
+                .filter(memberShape -> containsSensitiveWord(memberShape.getMemberName())
+                        || containsSensitivePhrase(memberShape.getMemberName()))
+                .map(this::emit)
+                .collect(Collectors.toList());
```

Comment:
nit: It would be nice if these `scan` functions used a consistent methodology - either loops and a list to add events to OR streaming with collectors.